source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.cc

```diff
@@ -67,35 +77,64 @@ Http::FilterTrailersStatus Http1BridgeFilter::encodeTrailers(Http::ResponseTrail
   }
 
   if (do_bridging_) {
-    // Here we check for grpc-status. If it's not zero, we change the response code. We assume
-    // that if a reset comes in and we disconnect the HTTP/1.1 client it will raise some type
-    // of exception/error that the response was not complete.
-    const Http::HeaderEntry* grpc_status_header = trailers.GrpcStatus();
-    if (grpc_status_header) {
-      uint64_t grpc_status_code;
-      if (!absl::SimpleAtoi(grpc_status_header->value().getStringView(), &grpc_status_code) ||
-          grpc_status_code != 0) {
-        response_headers_->setStatus(enumToInt(Http::Code::ServiceUnavailable));
-      }
-      response_headers_->setGrpcStatus(grpc_status_header->value().getStringView());
-    }
-
-    const Http::HeaderEntry* grpc_message_header = trailers.GrpcMessage();
-    if (grpc_message_header) {
-      response_headers_->setGrpcMessage(grpc_message_header->value().getStringView());
-    }
-
-    // Since we are buffering, set content-length so that HTTP/1.1 callers can better determine
-    // if this is a complete response.
-    response_headers_->setContentLength(
-        encoder_callbacks_->encodingBuffer() ? encoder_callbacks_->encodingBuffer()->length() : 0);
+    // We're bridging, so we need to process trailers and set the http status, content length,
+    // grpc status, and grpc message from those trailers.
+    doResponseTrailers(trailers);
   }
 
   // NOTE: We will still write the trailers, but the HTTP/1.1 codec will just eat them and end
   //       the chunk encoded response which is what we want.
   return Http::FilterTrailersStatus::Continue;
 }
 
+void Http1BridgeFilter::updateHttpStatusAndContentLength(
+    const Http::ResponseHeaderOrTrailerMap& trailers, bool enable_http_status_codes) {
+  // Here we check for grpc-status. If it's not zero, we change the response code. We assume
+  // that if a reset comes in and we disconnect the HTTP/1.1 client it will raise some type
+  // of exception/error that the response was not complete.
+  ASSERT(response_headers_);
```

Comment:
This ASSERT won't harm, but `updateGrpcStatusAndMessage()` could benefit from it more.

Inside `updateHttpStatusAndContentLength()` `response_headers_` is dereferenced unconditionally, so a test would always crash successfully if the pointer was not initialized.

Reply:
oops! thank you for catching, the two fxns got switched in my head. Moved the assert to `updateGrpcStatusAndMessage()`